### ISSUE TITLE
`IncomingRequest::getVar('key')` casts some JSON value to string

### ISSUE THREAD

**Issue by kenjis** [codeigniter4/CodeIgniter4#6878]
`int` and `bool` values are cast to `string`.
Is this intended behavior or a bug?

The following test passes:
```php
    public function testGetVarAndGetVarSpecificItem()
    {
        $data = [
            'str'   => 'str',
            'int'   => 123,
            'null'  => null,
            'true'  => true, 'false' => false,
            'array' => ['a', 1],
            'obj'   => (object) ['a' => 'a', 'b' => 123],
        ];
        $json = json_encode($data);

        $config          = new App();
        $config->baseURL = 'http://example.com/';

        $request = new IncomingRequest($config, new URI(), $json, new UserAgent());
        $request->setHeader('Content-Type', 'application/json');

        $all = $request->getVar();

        $expected = (object) $data;
        $this->assertEqualsCanonicalizing($expected, $all);

        $this->assertSame('str', $request->getVar('str'));
        // converted to string.
        $this->assertSame('123', $request->getVar('int'));
        $this->assertNull($request->getVar('null'));
        // converted to string.
        $this->assertSame('1', $request->getVar('true'));
        // converted to string.
        $this->assertSame('', $request->getVar('false'));
        $this->assertSame(['a', 1], $request->getVar('array'));
        $this->assertEqualsCanonicalizing(
            (object) ['a' => 'a', 'b' => 123],
            $request->getVar('obj')
        );
    }
```

**Comment by michalsn** [comment]
For me, this is a bug or actually two bugs.

To be honest, I  don't really like this feature of `getVar()`. It is supposed to be convenient, but in fact, it brings so much inconsistency when we use it with JSON data...

1. `getVar(null)` will call `getJSON()` which will return all the data without filtering.
2. `getVar('index')` will be filtered, but... it seems like `getJsonVar()` doesn't perform filtering on nested values (arrays, objects), only on single variables.

This is very problematic: 
* some variables may not be filtered at all (arrays etc.), even if we set a proper flag - it will be ignored
* if we call a single variable, it will be converted to a string due to the nature of the `filter_var()` function
* basically, we can get different variable types depending on how we ask for them

So... the only solution I see is to check the variable type before `filter_var()` and change it back after filtering via `settype()` if no `false` value is returned. I'll try to prepare a PR.

**Comment by kenjis** [comment]
Closed by #6954
